lib/govspeak.rb

```diff
@@ -1,3 +1,4 @@
+require "active_support"
```

Comment:
I couldn't understand from the commit why this change was needed.

Reply:
@kevindew Fair shout – I've just broken this out into its own commit to document the need: 8dbdcf8aa5487672ad70f0977325574f3ce995ce

Reply:
Great thanks. Does that mean we don't need the next 2 requires? presumably active_support pulls them in?

Reply:
As I understand it, no:

```ruby
require "active_support"
```

just pulls in the bare minimum dependencies required to cherry-pick the features you want.

Whereas:

```ruby
require "active_support/all"
```
is what you'd use if you wanted to pull in the whole kitchen sink.

At least that's how I'm interpreting it based on the docs – see: [1.1.1 Cherry-picking a Definition](https://guides.rubyonrails.org/active_support_core_extensions.html#cherry-picking-a-definition) vs [1.1.4 Loading All Active Support](https://guides.rubyonrails.org/active_support_core_extensions.html#loading-all-active-support)

---

Somewhat concerningly, the test suite seems to pass even when I remove/comment out all the `active_support` imports entirely 😳 – i.e. this passes:

```ruby
# require "active_support"
# require "active_support/core_ext/hash"
# require "active_support/core_ext/array"
```

But it seems cherry-picking without the initial `active_support` include is where things break.

Reply:
Oh cool TIL - yes that makes sense.

As for your concern, as I understand govuk_publishing_components pulls in various activesupport bits, so they probably just get satisfied a few lines later. All a bit clunky compared to JS/Python